scripts/fabricstat

```diff
@@ -122,7 +122,10 @@ class FabricPortStat(FabricStat):
 
         table = []
         header = None
-        asic = multi_asic.get_asic_id_from_name(self.namespace)
+        if not self.namespace:
```

Comment:
The problem we saw is that for single asic namespace is set to constants.DEFAULT_NAMESPACE, i.e. '',  and the command can not print out any information and with error of unknown asic namespace. "-n asic0" option also not working and errored out with following message because it can only get '' as namespace. 
```Invalid value for "--namespace" / "-n": invalid choice: asic0. (choose from )``

I think this command should be able to print out the the correct information for single asic case ( i.e. get the information for asic0 ) rather than errored out here. Let me know if you are ok with this.

Reply:
I am trying to reword about the piece of code here and see if I can explain it clear this time. This diff hunk is not used for filtering the show command output per ASIC, but rather to determine the ASIC name used in the first column of the output. `self.namespace` is an empty string for single chip systems. For multi-chip systems, `self.namespace` is set to the corresponding chip. To be clear the variable `asic` has been renamed to `asic_name` in the subsequent commit.

Reply:
For supervisor and Linecards with **multiple** chips , **with** and **without** my change the output looks like:
```
# show fabric counters port 
  ASIC    PORT    STATE    IN_CELL    IN_OCTET    OUT_CELL    OUT_OCTET    CRC    FEC_CORRECTABLE    FEC_UNCORRECTABLE    SYMBOL_ERR
------  ------  -------  ---------  ----------  ----------  -----------  -----  -----------------  -------------------  ------------
     0       0     up          0           0           0            0      0               2173           9884447862             0
     0       1     up          0           0           0            0      0         1454987235           8428978748             0
     0       2     up          0           0           0            0      0         1277815510           8566079276     124360072
     0       3     up          0           0           0            0      0                  0           9883123218             0
.....
    11     0       up          3         484           1          116      0                  2           2604843640             2
    11     1       up          2         251           0            0      0         1596791148             20286947    1560744220
    11     2       up          0           0           0            0      0                  0          10046118637             0
    11     3       up          0           0           0            0      0                  0          10045252879             0



# show fabric counters port -n asic0
  ASIC    PORT    STATE    IN_CELL    IN_OCTET    OUT_CELL    OUT_OCTET    CRC    FEC_CORRECTABLE    FEC_UNCORRECTABLE    SYMBOL_ERR
------  ------  -------  ---------  ----------  ----------  -----------  -----  -----------------  -------------------  ------------
     0       0     up          0           0           0            0      0               2398          10557594615             0
     0       1     up          0           0           0            0      0         1525593297           9031519953             0
     0       2     up          0           0           0            0      0         1356633584           9160407979     124360072
     0       3     up          0           0           0            0      0                  0          10556270609             0
 
```


For Linecards with only **one** chip, the output **without** my change looks like:

```
# show fabric counters port
Traceback (most recent call last):
  File "/usr/local/bin/fabricstat", line 302, in <module>
    main()
  File "/usr/local/bin/fabricstat", line 296, in main
    nsStat(ns, errors_only)
  File "/usr/local/bin/fabricstat", line 290, in nsStat
    stat.cnstat_print(cnstat_dict, errors_only)
  File "/usr/local/bin/fabricstat", line 130, in cnstat_print
    asic_name = multi_asic.get_asic_id_from_name(self.namespace)
  File "/usr/local/lib/python3.9/dist-packages/sonic_py_common/multi_asic.py", line 127, in get_asic_id_from_name
    raise ValueError('Unknown asic namespace name {}'.format(asic_name))
ValueError: Unknown asic namespace name

# show fabric counters port -n asic0
Usage: show fabric counters port [OPTIONS]
Try "show fabric counters port -h" for help.

Error: Invalid value for "--namespace" / "-n": invalid choice: asic0. (choose from )
```
For Linecards with only **one** chip, the output **with** my change looks like:

```
# show fabric counters port
  ASIC    PORT    STATE    IN_CELL    IN_OCTET    OUT_CELL    OUT_OCTET    CRC    FEC_CORRECTABLE    FEC_UNCORRECTABLE    SYMBOL_ERR
------  ------  -------  ---------  ----------  ----------  -----------  -----  -----------------  -------------------  ------------
     0       0       up          3         431           2          363      0                145               202358           135
     0       1       up          2         251           5          811      0                158               289742           148
     0       2       up          2         251           3          375      0                 47               205082            37
     0       3       up          3         431           2          362      0                117               257840           117


# show fabric counters port -n asic0
Usage: show fabric counters port [OPTIONS]
Try "show fabric counters port -h" for help.

Error: Invalid value for "--namespace" / "-n": invalid choice: asic0. (choose from )
```

Reply:
Thanks @jfeng-arista for sharing the output. Can you update the PR description with this output.

Reply:
Updated. thank you